src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Krivodonova.hpp

```diff
@@ -0,0 +1,1002 @@
+ *
+ * - limit \f$c_{2,2,2}\f$ (i.e. \f$c_{2,2,2}\leftarrow\tilde{c}_{2,2,2}\f$), if
+ *   not changed, stop
+ * - limit \f$c_{2,2,1}\f$, \f$c_{2,1,2}\f$, and \f$c_{1,2,2}\f$, if all not
+ * changed, stop
+ * - limit \f$c_{2,2,0}\f$, \f$c_{2,0,2}\f$, and \f$c_{0,2,2}\f$, if all not
+ * changed, stop
+ * - limit \f$c_{2,1,1}\f$, \f$c_{1,2,1}\f$, and \f$c_{1,1,2}\f$, if all not
+ * changed, stop
+ * - limit \f$c_{2,1,0}\f$, \f$c_{2,0,1}\f$, \f$c_{1,2,0}\f$, \f$c_{1,0,2}\f$,
+ * \f$c_{0,1,2}\f$, and \f$c_{0,2,1}\f$, if all not changed, stop
+ * - limit \f$c_{2,0,0}\f$, \f$c_{0,2,0}\f$, and \f$c_{0,0,2}\f$, if all not
+ * changed, stop
+ * - limit \f$c_{1,1,1}\f$, if not changed, stop
+ * - limit \f$c_{1,1,0}\f$, \f$c_{1,0,1}\f$, and \f$c_{0,1,1}\f$, if all not
+ * changed, stop
+ * - limit \f$c_{1,0,0}\f$, \f$c_{0,1,0}\f$, and \f$c_{0,0,1}\f$, if all not
+ * changed, stop
+ *
+ * The 1d and 2d implementations are straightforward restrictions of the
+ * described algorithm.
+ *
+ * #### Limitations:
+ *
+ * - We currently recompute the spectral coefficients for local use, after
+ *   having computed these same coefficients for sending to the neighbors. We
+ *   should be able to avoid this either by storing the coefficients in the
+ *   DataBox or by allowing the limiters' `packaged_data` function to return an
+ *   object to be passed as an additional argument to the `operator()` (still
+ *   would need to be stored in the DataBox).
+ * - We cannot handle the case where neighbors have more/fewer
+ *   coefficients. In the case that the neighbor has more coefficients we could
+ *   just ignore the higher coefficients. In the case that the neighbor has
+ *   fewer coefficients we have a few choices.
+ * - Having a different number of collocation points in different directions is
+ *   not supported. However, it is straightforward to handle this case. The
+ *   outermost loop should be in the direction with the most collocation points,
+ *   while the inner most loop should be over the direction with the fewest
+ *   collocation points. The highest to lowest coefficients can then be limited
+ *   appropriately again.
+ * - h-refinement is not supported, but there is one reasonably straightforward
+ *   implementation that may work. In this case we would ignore refinement that
+ *   is not in the direction of the neighbor, treating the element as
+ *   simply having multiple neighbors in that direction. The only change would
+ *   be accounting for different refinement in the direction of the neighor,
+ *   which should be easy to add since the differences in coefficients in
+ *   Eq.\f$\ref{eq:krivodonova 3d minmod}\f$ will just be multiplied by
+ *   non-unity factors.
+ */
+template <size_t VolumeDim, typename... Tags>
+class Krivodonova<VolumeDim, tmpl::list<Tags...>> {
+ public:
+  /*!
+   * \brief The \f$\alpha_i\f$ values in the Krivodonova algorithm.
+   *
+   * The default size is the maximum number of allowed Legendre basis
```

Comment:
"The default is 1. for all \f$\alpha_i\f$" and add that it always needs the maximum number of alphas